crates/autopilot/src/arguments.rs

```diff
@@ -298,6 +303,51 @@ impl std::fmt::Display for Arguments {
     }
 }
 
+#[derive(clap::Parser, Clone)]
+pub struct FeePolicy {
+    /// How much of the order's price improvement over max(limit price,
+    /// best_bid) should be taken as a protocol fee.
+    #[clap(
+        long,
+        env,
+        default_value = None,
+        value_parser = shared::arguments::parse_percentage_factor
+    )]
+    pub price_improvement_factor: Option<f64>,
+
+    /// How much of the order's volume should be taken as a protocol fee.
+    #[clap(
+        long,
+        env,
+        default_value = None,
+        value_parser = shared::arguments::parse_percentage_factor
+    )]
+    pub volume_factor: Option<f64>,
```

Comment:
Are you planning to use this as the cap if set? In this case I would make it clear that this is the max_volume_factor

Reply:
It's used for both capping of quote deviation fee and also for volume based fee. Capping actually means having another fee and choose the smaller between them. You can check the db PR where I use the same logic.

Reply:
Refactored to enum.

Reply:
Actually, with the refactoring to enum, it now seems to be resolved.